drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -244,23 +246,20 @@ static int pwm_nrf5_sw_init(struct device *dev)
  * channels 14-15 are used by BLE controller.
  */
 static const struct pwm_config pwm_nrf5_sw_0_config = {
-#if defined(CONFIG_SOC_SERIES_NRF51X)
-	.timer = NRF_TIMER1,
-	.ppi_base = 7,
-#else
-	.timer = NRF_TIMER2,
-	.ppi_base = 14,
-#endif
-	.gpiote_base = 0,
+	.timer = _CONCAT(NRF_TIMER, DT_NORDIC_NRF_SW_PWM_SW_PWM_TIMER_INSTANCE),
+	.ppi_base = DT_NORDIC_NRF_SW_PWM_SW_PWM_PPI_BASE,
```

Comment:
This is a single instance driver, so you can use `DT_NORDIC_NRF_SW_PWM_0_*` macros. They are more readable, and shorter.

Reply:
Changed to use the new DT defines.